components/tikv_util/src/logger/mod.rs

```diff
@@ -31,13 +33,37 @@ pub fn init_log<D>(
     level: Level,
     use_async: bool,
     init_stdlog: bool,
+    mut disabled_targets: HashSet<String>,
 ) -> Result<(), SetLoggerError>
 where
     D: Drain + Send + 'static,
     <D as Drain>::Err: std::fmt::Display,
 {
+    if let Ok(extra_modules) = env::var("TIKV_DISABLE_LOG_TARGETS") {
+        disabled_targets.extend(extra_modules.split(',').map(ToOwned::to_owned));
+    }
+
+    let filtered = drain.filter(move |record| {
+        if disabled_targets.is_empty() {
+            // The format of the returned value from module() would like this:
+            // ```
+            //  tikv::raftstore::store::fsm::store
+            //  tikv_util
+            //  tikv_util::config::check_data_dir
+            //  raft::raft
+            //  grpcio::log_util
+            //  ...
+            // ```
+            // Here get the highest level module name to check.
+            let module = record.module().splitn(2, "::").nth(0).unwrap();
+            !disabled_targets.contains(module)
```

Comment:
Maybe we can simply filter by `start_with("tokio_core")`

Reply:
for `tikv` and `tikv_util`, using `start_with` would be wrong.

Reply:
Looks like you can always filter by `starts_with()` something. For example, if you really want to filter away `tikv` but not `tikv_util`, you can use `tikv::`.

Reply:
The returned value of `record.module()` maybe `tikv` without `::`, so `tikv::` is also wrong.

Reply:
So when will you have a `record.module()` to be `tikv` and when you want to filter away logs from `tikv`? If this is not a scenario that is going to be happen, then I think it don't need to be considered.

Reply:
`tikv` is just an example, maybe other modules.